src/Nethermind/Nethermind.Blockchain/Synchronization/BlockDownloader.cs

```diff
@@ -200,8 +206,9 @@ public async Task<long> DownloadBlocks(PeerInfo bestPeer, int newBlocksToSkip, C
                         blocks[i - 1] = new Block(headers[i], BlockBody.Empty);
                     }
                 }
-                
-                Task<BlockBody[]> bodiesTask = hashes.Count == 0 ? Task.FromResult(new BlockBody[0]) : bestPeer.SyncPeer.GetBlocks(hashes.ToArray(), cancellation);
+
+                var blockHashes = hashes.ToArray();
```

Comment:
huge allocs

Reply:
it was like this before, interestingly..

Reply:
Removing with IList